notifications.py

```diff
@@ -344,6 +346,86 @@ def rule_process_groups_expiration_date(ctx):
     log.write(ctx, 'group expiration date - Finished checking research groups for reaching group expiration date | notified: {}'.format(notify_count))
 
 
+@rule.make()
+def rule_process_inactive_research_groups(ctx):
+    """Rule interface for checking for research groups that have not been modified after a certain amount of months.
+
+    :param ctx: Combined type of a callback and rei struct
+    """
+    # Only send notifications if inactivity notifications are enabled.
+    if config.inactivity_notification == 0:
+        return
+
+    # check permissions - rodsadmin only
+    if user.user_type(ctx) != 'rodsadmin':
+        log.write(ctx, "inactive research group - Insufficient permissions - should only be called by rodsadmin")
+        return
+
+    log.write(ctx, 'inactive research group - Checking Research packages for last modification dates')
+
+    zone = user.zone(ctx)
+    notify_count = 0
+    inactivity_cutoff = datetime.now() - timedelta(weeks=4.35 * config.inactivity_cutoff_months)
+    inactivity_cutoff_epoch = int((inactivity_cutoff - datetime(1970, 1, 1)).total_seconds())
+
+    # First query: obtain a list of groups with group attributes
+    iter = genquery.row_iterator(
+        "USER_GROUP_NAME",
+        "USER_TYPE = 'rodsgroup' AND USER_GROUP_NAME like 'research-%'",
+        genquery.AS_LIST, ctx
+    )
+
+    for row in iter:
+        group_name = row[0]
+        coll = '/{}/home/{}'.format(zone, group_name)
+        coll_general = '/{}/home/{}%'.format(zone, group_name)
```

Comment:
This wildcard matches subcollections (which is good) but also collections with the same prefix. For example, if you search for data objects in `/tempZone/home/research-foo%`, you will also get data objects in `/tempZone/home/research-foot`. Here's an example of code that matches with subcollections, but not other collections with the same prefix: https://github.com/UtrechtUniversity/yoda-ruleset/blob/a781dd81240236d851652b8b01d6ef94ee53b116/resources.py#L564-L579 .